backend/src/routes/classroom-group.ts

```diff
@@ -4,11 +4,122 @@ import { sql } from '@/lib/db';
 import { generateSlug } from '@/lib/utils';
 import { middleware } from '@/src/middleware';
 
-export const classroomGroupRoute = new Elysia({ prefix: '/c' }).group(
-    '/:slug/g',
-    (app) =>
+export const classroomGroupRoute = new Elysia({ prefix: '/c' })
+    .use(middleware)
+    .ws('/:group_slug/g/chat', {
+        async open(ws) {
+            const {
+                user,
+                session,
+                params: { group_slug: groupSlug },
+            } = ws.data;
+
+            if (!user || !session) {
+                ws.send({
+                    status: 'error',
+                    message: 'Unauthenticated, Please sign in and try again',
+                });
+                ws.close();
+                return;
+            }
+
+            const chatHistory = await sql`
+            SELECT
+                content,
+                created_at,
+                created_by
+            FROM group_message
+            WHERE group_slug = ${groupSlug}
+            `;
```

Comment:
Broken Access Control, this will allow everyone to use this api. Check if
1. If user is a teacher, check if user's in the classroom
2. If user is a student, check if user's in the classroom that is in the classroom